src/solver/types.jl

```diff
@@ -427,10 +427,19 @@ the default configurations set in `default_config_dict()`.
 function AtmosConfig(config::Dict; comms_ctx = nothing)
     config = override_default_config(config)
     FT = config["FLOAT_TYPE"] == "Float64" ? Float64 : Float32
-    toml_dict = CP.create_toml_dict(
-        FT;
-        override_file = CP.merge_toml_files(config["toml"]),
-    )
+
+    # Check that config["toml"] is a filepath and the file does not exist
+    if config["toml"] isa AbstractString && !isfile(config["toml"])
+        @info "Could not find TOML file at path `$(joinpath(pwd(), config["toml"]))`"
+        toml_path_in_atmos_dir =
+            joinpath(dirname(@__FILE__), "..", "..", config["toml"])
+        if isfile(toml_path_in_atmos_dir)
+            @info "Using parameter file at `$toml_path_in_atmos_dir`"
+            config["toml"] = toml_path_in_atmos_dir
+        end
+    end
+    toml_dict = CP.create_toml_dict(FT; override_file = config["toml"])
```

Comment:
This is mutating `config`, which means that `AtmosConfig` should be `AtmosConfig!`! I think it's better that we try to keep each step immutable. Why can't we just put this code into the driver? It's not a lot of code, and `AtmosConfig` will work in a much simpler way. Then, instead of `dirname(@__FILE__)` we can use `pkgdir(CA)`

Reply:
Sorry, I misunderstood the earlier comment. I agree `AtmosConfig()` being in the driver would make things a lot simpler, but how would a user create an `AtmosConfig` struct without including the driver?